### PR TITLE
solution 17/07

### PR DESCRIPTION
**Pull request by VadimKadasiev**

### CODE REVIEW

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication auth = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(auth);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (MentorMateAcademy) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]: Add these to constant

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package book.store.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/dto/UserLoginRequestDto.java]:
```suggestion
        @NotBlank
```

File: src/main/java/book/store/dto/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        @Length(min = 8, max = 20)
+        @Email
+        String email,
+
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/dto/UserLoginRequestDto.java]: same

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication auth = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(auth);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HEADER_TYPE);
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String HEADER_TYPE = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/book/store/security/JwtAuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(auth);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HEADER_TYPE);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_TYPE)) {
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/book/store/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(TOKEN_TYPE.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**